src/common/mempool.cc

```diff
@@ -95,9 +97,21 @@ size_t mempool::pool_t::allocated_items() const
 
 void mempool::pool_t::adjust_count(ssize_t items, ssize_t bytes)
 {
-  thread_shard_index = (thread_shard_index == num_shards) ? pick_a_shard_int() : thread_shard_index;
-  shard[thread_shard_index].items += items;
-  shard[thread_shard_index].bytes += bytes;
+#ifndef _GNU_SOURCE
+  // fallback for lack of sched_getcpu()
+  const size_t shard_index = []() {
+    if (thread_shard_index == num_shards) {
+      thread_shard_index = pick_a_shard_int();
+    }
+    return thread_shard_index;
+  }();
+#else
+  // the expected path: we alway pick the shard for a cpu core
+  // a thread is executing on.
+  const size_t shard_index = pick_a_shard_int();
+#endif
+  shard[shard_index].items += items;
+  shard[shard_index].bytes += bytes;
```

Comment:
also shard[num_shards] where num_shards = 1 << num_shard_bits is 32 elements array, still has highcore issue if crimson run over 32 cores.

Reply:
I suppose 2^7 (128) might be a reasonable number for crimson test cases.

Reply:
see https://github.com/ceph/ceph/pull/53130/commits/b1803d7c1c6688723cc6a03d35d91522fad28b8c

Reply:
See https://github.com/ceph/ceph/pull/53130#discussion_r1400154784. I have to revert the change.